modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/OperationBeanDefinitionParser.java

```diff
@@ -0,0 +1,191 @@
+import org.mule.enricher.MessageEnricher;
+import org.mule.extension.introspection.DataQualifier;
+import org.mule.extension.introspection.DataType;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.introspection.Parameter;
+import org.mule.module.extension.internal.introspection.BaseDataQualifierVisitor;
+import org.mule.module.extension.internal.util.NameUtils;
+import org.mule.util.ArrayUtils;
+
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.MutablePropertyValues;
+import org.springframework.beans.PropertyValue;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.RuntimeBeanReference;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.ManagedList;
+import org.springframework.beans.factory.support.ManagedMap;
+import org.springframework.beans.factory.xml.BeanDefinitionParser;
+import org.springframework.beans.factory.xml.ParserContext;
+import org.springframework.util.xml.DomUtils;
+import org.w3c.dom.Element;
+
+/**
+ * A {@link BeanDefinitionParser} to parse message processors
+ * which execute operations implemented through the extensions API.
+ * <p/>
+ * It defines an {@link OperationFactoryBean} which in turn builds
+ * the actual {@link MessageProcessor}
+ *
+ * @since 3.7.0
+ */
+final class OperationBeanDefinitionParser implements BeanDefinitionParser
+{
+
+    private final Operation operation;
+
+    OperationBeanDefinitionParser(Operation operation)
+    {
+        this.operation = operation;
+    }
+
+    @Override
+    public BeanDefinition parse(Element element, ParserContext parserContext)
+    {
+        BeanDefinitionBuilder builder = BeanDefinitionBuilder.rootBeanDefinition(OperationFactoryBean.class);
+        builder.setScope(BeanDefinition.SCOPE_PROTOTYPE);
+        parseConfigRef(element, builder);
+        builder.addConstructorArgValue(operation);
+        builder.addConstructorArgValue(toElementDescriptorBeanDefinition(element));
+        builder.addConstructorArgValue(parseNestedOperations(element, parserContext));
+        builder.addConstructorArgReference(MuleProperties.OBJECT_MULE_CONTEXT);
+
+        BeanDefinition definition = builder.getBeanDefinition();
+        setNoRecurseOnDefinition(definition);
+        attachProcessorDefinition(parserContext, definition);
+
+        return definition;
+    }
+
+    private ManagedMap<String, ManagedList<MessageProcessor>> parseNestedOperations(final Element element, final ParserContext parserContext)
+    {
+        final ManagedMap<String, ManagedList<MessageProcessor>> nestedOperations = new ManagedMap<>();
+
+        for (final Parameter parameter : operation.getParameters())
+        {
+            final DataType type = parameter.getType();
+            type.getQualifier().accept(new BaseDataQualifierVisitor()
```

Comment:
Can? Sure... Adds value? Don't think so.

To unit test this would require a lot of mocking and simulating XML parsers and I simply don't see the value, on the contrary, the code would become harder to follow.

IMHO, The best way to test this code (and the way we generally test parsers out in Mule) is to simply have test configurations that account for these cases and verified they're parsed ok. Since we already have coverage on that, I don't see much value in changing it.